test/unit/org/apache/cassandra/locator/Ec2SnitchTest.java

```diff
@@ -71,30 +124,60 @@ public static void setup() throws Exception
         StorageService.instance.initServer(0);
     }
 
-    private class TestEC2Snitch extends Ec2Snitch
+    private static class TestEc2Snitch extends Ec2Snitch
     {
-        public TestEC2Snitch() throws IOException, ConfigurationException
+        public TestEc2Snitch() throws IOException, ConfigurationException
         {
-            super();
+            this(new SnitchProperties());
         }
 
-        public TestEC2Snitch(SnitchProperties props) throws IOException, ConfigurationException
+        public TestEc2Snitch(SnitchProperties props) throws IOException, ConfigurationException
```

Comment:
What do you think about refactoring this test a bit? We have separated a snitch from a connector so it would be enough to simply mock the connector and do not bother distinguishing between v1 and v2 at all. The whole test would be much simpler and easier to follow